utils/odb-tools.ts

```diff
@@ -118,15 +138,24 @@ function eth_signTypedData(primaryType, message, domain, wallet) {
     });
 }
 
-function signStruct(primaryType, message, domain, wallet) {
+export async function signStruct(
+    primaryType: string,
+    message: Record<string, any>,
+    domain: TypedDataDomain,
+    wallet: WalletInfo,
+): Promise<Record<string, any>> {
     return eth_signTypedData(primaryType, message, domain, wallet).then((sign) => {
         message.sign = sign;
         return message;
     });
 }
 
-function hashStruct(primaryType, message, domain) {
-    let typedDataDomain = {
+export function hashStruct(
+    primaryType: string,
+    message: Record<string, any>,
+    domain: TypedDataDomain,
+): string {
+    const typedDataDomain: TypedDataDomain = {
```

Comment:
```suggestion
    const typedDataDomain = {
```
This is a self-inferred type. It's useless to specify it. But we can keep it if you want